compiler/rustc_infer/src/traits/error_reporting/mod.rs

```diff
@@ -103,9 +103,11 @@ pub fn report_object_safety_error<'tcx>(
     if trait_span.is_some() {
         let mut reported_violations: Vec<_> = reported_violations.into_iter().collect();
         reported_violations.sort();
+
+        let mut reported_names = vec![];
```

Comment:
Could you add a little comment explaining why this is needed like "allows us to skip suggesting moving the same item to another trait multiple times" or something?

Reply:
Done! Do you think there is a better way of doing this to hide the vector, maybe by taking in the list of violations up front for example or is it okay as is?

Reply:
We could create an `ObjectSafetyViolationSolution` enum, which would be returned from `solution`, then deduplicated, then added to the error. Something like
```rust
let mut potential_solutions: Vec<_> = reported_violations.into_iter().map(...::solution).collect();
potential_solutions.sort();
potential_solutions.dedup();
potential_solutions.for_each(|solution| solution.add_to(&mut err));
```

Do you want to try doing this?

Reply:
Ah I see, that sounds better, I'll give it a try!